drv/i2c-devices/src/mwocp68.rs

```diff
@@ -62,6 +117,66 @@ impl From<pmbus::Error> for Error {
     }
 }
 
+///
+/// Defines the state of the firmware update.  Once `UpdateSuccessful`
+/// has been returned, the update is complete.
+///
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum UpdateState {
+    /// The boot loader key has been written
+    WroteBootLoaderKey,
+
+    /// The product key has been written
+    WroteProductKey,
+
+    /// The boot loader has been booted
+    BootedBootLoader,
+
+    /// Programming of firmware has been indicated to have started
+    StartedProgramming,
+
+    /// A block has been written; the next offset is at [`offset`], and the
+    /// running checksum is in [`checksum`]
+    WroteBlock { offset: usize, checksum: u64 },
```

Comment:
Nit: it looks like we only use a `u16` for the checksum, so we could store a `u16` and use `wrapping_add` (instead of unchecked arithmetic and trusting that we'll never overflow a `u64`).

Reply:
I don't want to chuck the higher bits of the checksum even though the part only wants the low 16 -- but agreed on the unchecked arithmetic; what about using a `wrapping_add` on the checksum?

Reply:
Yeah, sounds good